### PR TITLE
Feature/network anomaly

### PR DESCRIPTION
**Pull request by amitschendel**

### CODE REVIEW

File: pkg/ruleengine/v1/factory.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ func NewRuleCreator() *RuleCreatorImpl {
 			R0008ReadEnvironmentVariablesProcFSRuleDescriptor,
 			R0009EbpfProgramLoadRuleDescriptor,
 			R0010UnexpectedSensitiveFileAccessRuleDescriptor,
+			R0011UnexpectedNetworkTrafficRuleDescriptor,
```
Comment (slashben) [inline pkg/ruleengine/v1/factory.go:24]: I would call this rather "Unexpected Egress Network Traffic", we are detecting only egress.(I am not planning internal traffic, even if so, in a different rule)

File: pkg/ruleengine/v1/r0011_unexpected_network_traffic.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+	"github.com/kubescape/node-agent/pkg/objectcache"
+	"github.com/kubescape/node-agent/pkg/ruleengine"
+	"github.com/kubescape/node-agent/pkg/utils"
+
+	tracernetworktype "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/network/types"
+)
+
+const (
+	R0011ID   = "R0011"
+	R0011Name = "Unexpected network traffic"
```
Comment (slashben) [inline pkg/ruleengine/v1/r0011_unexpected_network_traffic.go:21]: Same applicable here :)

**Review by matthyx — APPROVED** [review]

File: pkg/ruleengine/v1/r0011_unexpected_network_traffic.go [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+	// Check if IP is in private IP ranges
+	privateIPRanges := []struct {
+		start net.IP
+		end   net.IP
+	}{
+		{net.ParseIP("10.0.0.0"), net.ParseIP("10.255.255.255")},
+		{net.ParseIP("172.16.0.0"), net.ParseIP("172.31.255.255")},
+		{net.ParseIP("192.168.0.0"), net.ParseIP("192.168.255.255")},
+	}
```
Comment (slashben) [inline pkg/ruleengine/v1/r0011_unexpected_network_traffic.go:164]: @matthyx , do you agree with these ranges? All pod network IPs should be covered with these ranges

File: pkg/ruleengine/v1/r0011_unexpected_network_traffic.go [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+func (rule *R0011UnexpectedNetworkTraffic) handleNetworkEvent(networkEvent *tracernetworktype.Event, objCache objectcache.ObjectCache) ruleengine.RuleFailure {
+	// Check if we already alerted on this address.
+	if ok := rule.alertedAdresses.Has(networkEvent.DstEndpoint.Addr); ok {
+		return nil
+	}
+
+	// Check if the network event is outgoing and the destination is not a private IP.
+	if networkEvent.PktType == "OUTGOING" && !isPrivateIP(networkEvent.DstEndpoint.Addr) {
```
Comment (slashben) [inline pkg/ruleengine/v1/r0011_unexpected_network_traffic.go:65]: Is UDP covered here?

File: pkg/ruleengine/v1/r0011_unexpected_network_traffic.go [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+			}
+
+			// Check if we seen this dns name before and it's in-cluster address and in the egress list.
+			if domain != "" && (strings.HasSuffix(domain, "svc.cluster.local.") || slices.Contains(egress.DNSNames, domain)) {
+				return nil
+			}
+		}
+
+		// Alert on the address.
+		rule.alertedAdresses.Set(networkEvent.DstEndpoint.Addr, true)
```
Comment (slashben) [inline pkg/ruleengine/v1/r0011_unexpected_network_traffic.go:91]: Shouldn't this happen per pod/container?

File: pkg/ruleengine/v1/r0011_unexpected_network_traffic.go (hunk 3 above)
Comment (matthyx) [reply pkg/ruleengine/v1/r0011_unexpected_network_traffic.go:164]: yes, we should include ipv6 at a later point

File: pkg/ruleengine/v1/r0011_unexpected_network_traffic.go (hunk 5 above)
Comment (amitschendel) [reply pkg/ruleengine/v1/r0011_unexpected_network_traffic.go:91]: The rules are at the container level all ready no?

File: pkg/ruleengine/v1/r0011_unexpected_network_traffic.go (hunk 4 above)
Comment (amitschendel) [reply pkg/ruleengine/v1/r0011_unexpected_network_traffic.go]: For some reason UDP is currently now working, need to investigate. Is that a blocker?

File: pkg/ruleengine/v1/r0011_unexpected_network_traffic.go (hunk 2 above)
Comment (amitschendel) [reply pkg/ruleengine/v1/r0011_unexpected_network_traffic.go]: done

File: pkg/ruleengine/v1/factory.go (hunk 1 above)
Comment (amitschendel) [reply pkg/ruleengine/v1/factory.go]: done

File: pkg/ruleengine/v1/r0011_unexpected_network_traffic.go (hunk 5 above)
Comment (slashben) [reply pkg/ruleengine/v1/r0011_unexpected_network_traffic.go]: my bad

File: pkg/ruleengine/v1/r0011_unexpected_network_traffic.go (hunk 4 above)
Comment (slashben) [reply pkg/ruleengine/v1/r0011_unexpected_network_traffic.go]: no

**Review by slashben — APPROVED** [review]

**Pull request merged**